sled-agent/src/services.rs

```diff
@@ -4940,6 +4940,7 @@ mod illumos_tests {
         zone::MockZones,
     };
 
+    use nexus_sled_agent_shared::inventory::OmicronZoneImageSource;
```

Comment:
I'm all for having this PR not try to also fix sled agent to honor this field at the same time, but what do you think about having it reject any use of `OmicronZoneImageSource::Artifact` in the meantime?

Reply:
I think this might make it more difficult to test the Nexus side of it if `omicron_zones_put` is rejecting anything using the Artifact variant, and I'm not sure I want to figure out how to word and wire up a new error variant to rip out in an upcoming PR.